### PR TITLE
chore/remove class files, fix typos, code refactor and sonar findings

### PR DESCRIPTION
**Pull request by nitin-vavdiya**
## Description
1. Class files removed 
2. Typos fixed in exception classes 
3. Java doc fixed in exception classes 
4. code refactor 
5. Sonar finding

Before sonar fixes:
Screenshot 2024-02-26 at 12 00 20 PM

After sonar fixes:
Screenshot 2024-02-26 at 12 38 24 PM

## Pre-review checks

Please ensure to do as many of the following checks as possible, before asking for committer review:

- [x] DEPENDENCIES are up-to-date. [Dash license tool](https://github.com/eclipse/dash-licenses). Committers can open IP issues for restricted libs.
- [x] [Copyright and license header](https://eclipse-tractusx.github.io/docs/release/trg-7/trg-7-02) are present on all affected files

### CODE REVIEW

**Comment by borisrizov-zf** [comment]
@nitin-vavdiya seems that the dash license tool has some issues right now.

**Comment by nitin-vavdiya** [comment]
> @nitin-vavdiya seems that the dash license tool has some issues right now.
Thanks for checking out.
Can we re-run today and check if the issue is resolved?

**Review by DominikPinsel — APPROVED** [review]

**Comment by nitin-vavdiya** [comment]
@borisrizov-zf  May I know what is the blocker to merge this PR?

**Comment by borisrizov-zf** [comment]
I was actually hoping to merge the #77 first. It's just taking some time.

File: src/main/java/org/eclipse/tractusx/ssi/lib/jwt/SignedJwtVerifier.java [hunk 1]
```diff
@@ -61,16 +60,15 @@ public class SignedJwtVerifier {
...
-   * @throws DidParseException
-   * @throws SignatureException
-   * @throws DidResolverException
-   * @throws SignatureVerificationException
-   * @throws UnsupportedVerificationMethodException
-   * @throws SignatureParseException
+   * @throws DidParseException the did parse error
+   * @throws DidResolverException the did resolve error
+   * @throws SignatureVerificationException the signature verification error
+   * @throws UnsupportedVerificationMethodException the unsupported verification error
```
Comment (borisrizov-zf) [inline src/main/java/org/eclipse/tractusx/ssi/lib/jwt/SignedJwtVerifier.java:66]: This exception has been removed from the signature, but is still in the doc string.

File: src/main/java/org/eclipse/tractusx/ssi/lib/jwt/SignedJwtVerifier.java [hunk 2]
```diff
@@ -61,16 +60,15 @@ public class SignedJwtVerifier {
...
+   * @throws SignatureParseException the signature parse error
    */
   public boolean verify(SignedJWT jwt)
       throws DidParseException, DidResolverException, SignatureVerificationException,
-          UnsupportedVerificationMethodException, SignatureParseException {
+          SignatureParseException {
```
Comment (borisrizov-zf) [inline src/main/java/org/eclipse/tractusx/ssi/lib/jwt/SignedJwtVerifier.java:71]: Missing from doc string.

File: src/main/java/org/eclipse/tractusx/ssi/lib/model/JsonLdObject.java [hunk 3]
```diff
@@ -51,7 +51,7 @@ public abstract class JsonLdObject extends HashMap<String, Object> {
    *
    * @param json the json
    */
-  public JsonLdObject(Map<String, Object> json) {
+  protected JsonLdObject(Map<String, Object> json) {
```
Comment (borisrizov-zf) [inline src/main/java/org/eclipse/tractusx/ssi/lib/model/JsonLdObject.java:54]: Why is the visibility changed here?

File: src/main/java/org/eclipse/tractusx/ssi/lib/model/verifiable/Verifiable.java [hunk 4]
```diff
@@ -57,7 +57,7 @@ public enum VerifiableType {
    * @param json the json
    * @param type the type
    */
-  public Verifiable(Map<String, Object> json, VerifiableType type) {
+  protected Verifiable(Map<String, Object> json, VerifiableType type) {
```
Comment (borisrizov-zf) [inline src/main/java/org/eclipse/tractusx/ssi/lib/model/verifiable/Verifiable.java:60]: Why is the visibility changed?

File: src/main/java/org/eclipse/tractusx/ssi/lib/proof/IVerifier.java [hunk 5]
```diff
@@ -39,18 +38,16 @@ public interface IVerifier {
...
-   * @param document {@link VerifiableCredential} the verifiable
+   * @param verifiable {@link VerifiableCredential} the verifiable
    * @return boolean if verified or not
    * @throws UnsupportedSignatureTypeException
    * @throws SignatureParseException
    * @throws InvalidPublicKeyFormatException
-   * @throws SignatureGenerateFailedException
    * @throws SignatureVerificationException
    * @throws DidParseException
    * @throws NoVerificationKeyFoundException
```
Comment (borisrizov-zf) [inline src/main/java/org/eclipse/tractusx/ssi/lib/proof/IVerifier.java:48]: Seems unused.

File: src/main/java/org/eclipse/tractusx/ssi/lib/proof/IVerifier.java [hunk 6]
```diff
@@ -39,18 +38,16 @@ public interface IVerifier {
    * model to get the public key of issuer.
    *
    * @param hashedLinkedData the hashed linked data
-   * @param document {@link VerifiableCredential} the verifiable
+   * @param verifiable {@link VerifiableCredential} the verifiable
    * @return boolean if verified or not
    * @throws UnsupportedSignatureTypeException
```
Comment (borisrizov-zf) [inline src/main/java/org/eclipse/tractusx/ssi/lib/proof/IVerifier.java:43]: Seems unused.

File: src/test/java/org/eclipse/tractusx/ssi/lib/proof/SignAndVerifyTest.java [hunk 7]
```diff
@@ -34,19 +34,12 @@
...
 
-  /**
-   * Test sign and verify ed 201559.
-   *
-   * @throws IOException the io exception
-   * @throws InvalidePrivateKeyFormat the invalide private key format
-   * @throws InvalidePublicKeyFormat the invalide public key format
-   * @throws KeyGenerationException the key generation exception
-   */
+  /** Test sign and verify ed 201559. */
```
Comment (borisrizov-zf) [inline src/test/java/org/eclipse/tractusx/ssi/lib/proof/SignAndVerifyTest.java:39]:
```suggestion
  /** Test sign and verify ed 25519. */
```

File: src/test/java/org/eclipse/tractusx/ssi/lib/proof/SignAndVerifyTest.java [hunk 8]
```diff
@@ -34,19 +34,12 @@
...
   @Test
   @SneakyThrows
-  public void testSignAndVerify_ED201559() {
+  void testSignAndVerify_ED201559() {
```
Comment (borisrizov-zf) [inline src/test/java/org/eclipse/tractusx/ssi/lib/proof/SignAndVerifyTest.java:42]:
```suggestion
  void testSignAndVerify_ED25519() {
```

**Review by borisrizov-zf — CHANGES_REQUESTED** [review]

File: src/main/java/org/eclipse/tractusx/ssi/examples/VP.java [hunk 9]
```diff
@@ -22,18 +22,16 @@
...
-import org.eclipse.tractusx.ssi.lib.crypt.IPublicKey;
 import org.eclipse.tractusx.ssi.lib.crypt.octet.OctetKeyPairFactory;
 import org.eclipse.tractusx.ssi.lib.jwt.SignedJwtFactory;
 import org.eclipse.tractusx.ssi.lib.model.did.Did;
 import org.eclipse.tractusx.ssi.lib.model.verifiable.credential.VerifiableCredential;
 import org.eclipse.tractusx.ssi.lib.model.verifiable.presentation.VerifiablePresentation;
 import org.eclipse.tractusx.ssi.lib.model.verifiable.presentation.VerifiablePresentationBuilder;
 import org.eclipse.tractusx.ssi.lib.model.verifiable.presentation.VerifiablePresentationType;
-import org.eclipse.tractusx.ssi.lib.serialization.jsonLd.JsonLdSerializerImpl;
+import org.eclipse.tractusx.ssi.lib.serialization.jsonld.JsonLdSerializerImpl;
```
[the remark this answers is not in the document]
Comment (nitin-vavdiya) [reply src/main/java/org/eclipse/tractusx/ssi/examples/VP.java:34]: This is just to align with Java naming convention

File: src/main/java/org/eclipse/tractusx/ssi/lib/model/JsonLdObject.java (hunk 3 above)
Comment (nitin-vavdiya) [reply src/main/java/org/eclipse/tractusx/ssi/lib/model/JsonLdObject.java:54]: So far this object is accessed within package only, again this a sonar suggestion

File: src/main/java/org/eclipse/tractusx/ssi/lib/model/verifiable/Verifiable.java (hunk 4 above)
Comment (nitin-vavdiya) [reply src/main/java/org/eclipse/tractusx/ssi/lib/model/verifiable/Verifiable.java:60]: Same as above

File: src/main/java/org/eclipse/tractusx/ssi/lib/proof/IVerifier.java (hunk 6 above)
Comment (nitin-vavdiya) [reply src/main/java/org/eclipse/tractusx/ssi/lib/proof/IVerifier.java:43]: verifiable is a method argument of verify method

**Comment by nitin-vavdiya** [comment]
@borisrizov-zf  Class rename is only to align with Java naming standard

**Comment by nitin-vavdiya** [comment]
@borisrizov-zf  rebase done

**Comment by borisrizov-zf** [comment]
@nitin-vavdiya hi, I checked the rebase-ability of the branch against #91 . They have merge conflicts, so please wait for the merge and rebase. While you're rebasing, please also fix typos in the git commits such as in `ec99f9d11e87d4e4af35f6e3717ef53e752fc7e4` where you've typed `chroe` instead of `chore`. I'll ping you when you can start rebasing.

**Comment by borisrizov-zf** [comment]
@nitin-vavdiya Hi, the JWT branch is merged. You can go ahead and rebase. You'll get some issues, but it's mostly updated docstrings.

**Comment by nitin-vavdiya** [comment]
@borisrizov-zf  Rebase  done

**Pull request merged**